### PR TITLE
SDIT-1679 Add support for searching on multiple string values

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt [hunk 1]
```diff
@@ -45,32 +45,35 @@ class AttributeSearchResource(private val attributeSearchService: AttributeSearc
   @PostMapping
   @Tag(name = "Attribute search")
   @Operation(
-    summary = "*** WIP - DO NOT USE!!! *** Search for prisoners by attributes",
+    summary = "*** BETA *** Search for prisoners by attributes",
     description = """<p>This endpoint allows you to create queries over all attributes from the <em>Prisoner</em> record. Requires ROLE_GLOBAL_SEARCH or ROLE_PRISONER_SEARCH role.</p>
       <p>The request contains a list of queries to search on one or more attributes using a list of matchers. For example attribute "lastName""
       requires a <em>StringMatcher</em> so we can query on <strong>"lastName IS Smith"</strong>. Other type matchers include <em>IntMatcher</em>, <em>BooleanMatcher</em>,
-      <em>DateMatcher</em> and <em>DateTimeMatcher</em>. We have the facility to easily create additional matchers as required, for example 
-      we may end up with a PNCMatcher that handles any format of PNC number.
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt:53]: Tidied some of the swagger docs

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt [hunk 2]
```diff
@@ -45,32 +45,35 @@ class AttributeSearchResource(private val attributeSearchService: AttributeSearc
...
+      <em>DateMatcher</em> and <em>DateTimeMatcher</em>.
       </p>
       <p>Each query can also contain a list of sub-queries. Each sub-query can be considered as a separate query in brackets.
       Combining multiple sub-queries gives us the ability to create complex searches using any combination of a prisoner's 
-      attributes. For example we can model nested queries such as <strong>"lastName IS Smith AND (prisonId IS MDI OR (prisonId IS OUT AND inOutStatus is OUT))"</strong>.
+      attributes. For example we can model nested queries such as <strong>"lastName IS Smith AND (prisonId IS MDI OR (prisonId IS OUT AND lastPrisonId IS MDI))"</strong>.
       </p>
-      <p>Please be aware that when searching lists of complex objects (e.g. aliases, alerts, tattoos) if you want to search for multiple attributes within the same object then you need
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt:59]: Changed the order of these 2 paragraphs as it makes more sense.

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt [hunk 3]
```diff
@@ -112,6 +115,36 @@ class AttributeSearchResource(private val attributeSearchService: AttributeSearc
...
+                  "attribute": "prisonId",
+                  "condition": "IS",
+                  "searchTerm": "MDI"
+                },
+                {
+                  "type": "String",
+                  "attribute": "cellLocation",
+                  "condition": "IN",
+                  "searchTerm": "1-2-001,1-3-014,3-1-020"
+                }
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt:141]: The StringMatcher now supports searching for a list of values.

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**